src/onegov/core/types.py

```diff
@@ -52,3 +55,6 @@ class LaxFileDict(TypedDict):
 
     _T = TypeVar('_T')
     SequenceOrScalar = Union[Sequence[_T], _T]
+
+    class PaginatedGenericCollection(GenericCollection[_M], Pagination[_M]):
+        pass
```

Comment:
As seen in [ApisMixin](https://github.com/OneGov/onegov-cloud/blob/56e88649b900ef5eda6a8047b1f1d79ac843189b/src/onegov/agency/api.py#L35-L48) we use the following types:

 `PaginatedAgencyCollection` 
`ExtendedPersonCollection`
`PaginatedMembershipCollection`

They all have the thing in common (and this is expected from the implementation) that extend `GenericCollection` and `Pagination`, so i attempted to create `PaginatedGenericCollection` to imitates this. But I know I'm missing something here.
https://github.com/OneGov/onegov-cloud/blob/175c2b2b91528440709575c3882b04f45b826e49/src/onegov/api/models.py#L237

the last mypy error:
```bash
src/onegov/api/models.py:204: error: Argument 1 to "for_item" of "ApiEndpoint" has incompatible type "_M"; expected "ApiEndpointItem"  [arg-type]
```

Reply:
Yeah, you would need `Intersection` for that, so you can't really type this properly yet, unless you define a `Protocol` that matches this `Intersection` manually.